src/python/nimbusml/pipeline.py

```diff
@@ -819,6 +819,12 @@ def _fit_graph(self, X, y, verbose, **params):
                     model=output_predictor_model)
                 output_predictor_model_node._implicit = True
                 graph_nodes.append(output_predictor_model_node)
+        elif learner_node == None and transform_models[-1].startswith("$dataset_transformer_model"):
+            single_model_node = transforms_modelcombiner(
+                models=transform_models,
+                output_model=output_model)
+            single_model_node._implicit = True
+            graph_nodes.append(single_model_node)
```

Comment:
Will there be scenarios where a predictor or a non-DatasetTransformer transformer will come after a DatasetTransformer (which contains a predictor)?

Reply:
Yes, I have added this test case as well!